### ISSUE TITLE
Scroll listener is not removed if element is destroyed

### ISSUE THREAD

**Issue by macecyri** [aliok/ion-affix#1]
Thanks for this useful plugin!

I used the ion-affix directive on elements of a dynamic array (updated with ng-repeat) and the scroll listener (line 168: angular.element($ionicScroll.element).on('scroll', ....) is created each time my array is updated, resulting in strange behaviour because the scroll function is called twice after one update, three times after 2 updates etc...

I managed to resolve this problem by adding the folowing code to the directive 

```
$scope.$on("$destroy",function() {
                angular.element($ionicScroll.element).off('scroll');
            });
```

Could you add this snippet or a to the next release of the plugin? Or maybe you think there is a more elegant way to resolve this problem

Thanks again for your work

**Comment by aliok** [comment]
Thanks for reporting the problem and also suggesting the solution. I've added the fix you suggested.

**Comment by aliok** [comment]
Gonna release 1.0.1 soon with this fix.

**Comment by aliok** [comment]
Released 1.0.1.
